include/aws/s3/private/s3_client_impl.h

```diff
@@ -167,6 +167,25 @@ struct aws_s3_client_vtable {
         *parallel_input_stream_new_from_file)(struct aws_allocator *allocator, struct aws_byte_cursor file_name);
 };
 
+struct aws_s3_upload_part_timeout_stats {
+    bool stop_timeout;
+
+    /* Total number of upload request succeed */
+    size_t num_upload_requests_succeed;
+
+    /* Stats for the request time of first 10 succeed requests */
+    uint64_t request_time_sum_ns;
+    uint64_t number_request_time;
```

Comment:
debatable: name these so they're clearly related. substruct, or similar prefix.
```suggestion
    uint64_t request_time_sum_ns;
    uint64_t request_time_num_samples;
```
or
```suggestion
    struct {
        uint64_t sum_ns;
        uint64_t num_samples;
    } initial_request_time;
```